app/Main.hs

```diff
@@ -17,7 +63,7 @@ parseSubmissionResult :: (Int, BSL.ByteString) -> String
 parseSubmissionResult = BSL8.unpack . head . BSL8.split '.' . snd
 
 parseQstatResponse :: (Int, BSL.ByteString) -> String
-parseQstatResponse r = head $ tail $ reverse $ words $ head $ drop 2 $ lines $ BSL8.unpack $ snd r
+parseQstatResponse r = head $ tail $ reverse $ words $ (lines (BSL8.unpack $ snd r) !! 2)
```

Comment:
I'd love to discuss this in our next meeting 🤯

Reply:
Could you maybe add a comment about what's being returned and what the parsing steps do. An example string would make it easier to follow the code...

Reply:
See below